webfront/views/custom.py

```diff
@@ -78,9 +82,19 @@ def get(
         *args,
         **kwargs
     ):
+        
+        # Before any logic is executed, check correctness of accession format
+        db_members_accessions = r"^({})$".format("|".join((db["accession"] for (_, db) in settings.DB_MEMBERS.items())))
+        for i in range(0, len(endpoint_levels)):
+            if (re.match(db_members_accessions, endpoint_levels[i])):
+                if (not(re.match(settings.DB_MEMBERS[endpoint_levels[i-1]]["accession"], endpoint_levels[i]))):
+                    content = {"detail": f"{endpoint_levels[i]} is not a valid {endpoint_levels[i-1].upper()} accession"}
+                    return Response(content, status=status.HTTP_404_NOT_FOUND)
+
```

Comment:
A few points:

- It is more _pythonic_ to use `for i, my_item in enumerate(my_list)` instead of `for i in range(0, len(my_list))` followed by `my_list[i]`.
- By the way, `range(0, len(my_list))` can be simplified to `range(len(my_list))`.
- Nesting the check in `if level == 0` allows the check to run only once. Otherwise it runs for each level.
- Use `re.I` or `re.IGNORECASE` to perform case-insensitive searches.

```suggestion
        if level == 0:
            # Before any logic is executed, check correctness of accession format
            patterns = [db["accession"] for db in settings.DB_MEMBERS.values()]
            pattern = r"({})".format("|".join(patterns))
            regex = re.compile(pattern, re.IGNORECASE)

            for i, endpoint_level in enumerate(endpoint_levels):
                if regex.match(endpoint_level):
                    accession = endpoint_level.upper()
                    db_key = endpoint_levels[i - 1].lower()
                    db_member = settings.DB_MEMBERS[db_key]
                    pattern = db_member["accession"]
                    match = re.match(pattern, accession, re.IGNORECASE)
                    if match is None:
                        return Response(
                            {
                                "Error": f"{accession} is not a valid "
                                         f"{db_member['label']} accession",
                            },
                            status=status.HTTP_404_NOT_FOUND
                        )

```

Check you're happy with this suggestion, and if so, feel free to merge.

Reply:
Yup, all good points. Tested out multiple cases again and it works. Committing the changes and merging. Thanks.